src/averageAges.js

```diff
@@ -34,7 +50,26 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let counter = 0;
+  const totalAge = people.reduce((sum, person) => {
+    if (person.sex === 'f'
+      && withChildren
+      && people.some(potentialChild => potentialChild.mother === person.name)) {
+      counter++;
+
+      return sum + (person.died - person.born);
+    }
+
+    if (person.sex === 'f' && withChildren === undefined) {
+      counter++;
+
+      return sum + (person.died - person.born);
+    };
+
+    return sum;
+  }, 0);
```

Comment:
Same here. 'Filter' incoming array, then use 'some' and use the length of an array instead of 'counter'